api/handlers.go

```diff
@@ -412,6 +457,67 @@ func (api *API) trackRegistryWriteHandler(w http.ResponseWriter, req *http.Reque
 	api.WriteSuccess(w)
 }
 
+// skylinkDeleteHandler unpins all uploads of a skylink uploaded by the user.
+func (api *API) skylinkDeleteHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	sub, _, _, err := jwt.TokenFromContext(req.Context())
+	if err != nil {
+		api.WriteError(w, err, http.StatusUnauthorized)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub, false)
+	if errors.Contains(err, database.ErrUserNotFound) {
+		api.WriteError(w, err, http.StatusNotFound)
+		return
+	}
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	sl := ps.ByName("skylink")
+	if !database.ValidSkylinkHash(sl) {
+		api.WriteError(w, errors.New("invalid skylink"), http.StatusBadRequest)
+	}
+	skylink, err := api.staticDB.Skylink(req.Context(), sl)
+	if errors.Contains(err, database.ErrInvalidSkylink) {
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	_, err = api.staticDB.UnpinUploads(req.Context(), *skylink, *u)
+	if err != nil {
+		api.WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	api.WriteSuccess(w)
+	// Now that we've returned results to the caller, we can take care of some
+	// administrative details, such as user's quotas check.
+	// Nota that this call is not affected by the request's context, so we use
+	// a separate one.
+	api.checkUserQuotas(context.Background(), u)
+}
+
+// checkUserQuotas compares the resources consumed by the user to their quotas
+// and sets the QuotaExceeded flag on their account if they exceed any.
+func (api *API) checkUserQuotas(ctx context.Context, u *database.User) {
```

Comment:
We're calling this on delete/unpin but are we also calling it on upload?

Reply:
Absolutely, check out `trackUploadHandler`.